editor/plugins/canvas_item_editor_plugin.cpp

```diff
@@ -5538,7 +5538,9 @@ void CanvasItemEditorViewport::_remove_preview() {
 			node->queue_free();
 			preview_node->remove_child(node);
```

Comment:
```suggestion
```

(instead of the current change)

Because of https://github.com/godotengine/godot/blob/ce10ca69794900896a4162efc823386ce5bde3dd/scene/main/node.cpp#L220-L226 this line will (sometimes) cause `_remove_preview` to be re-invoked and removing the preview node, causing the issue.
If this is line removed the node will only be removed as a child when the queued deletion happens (after this method has completed, avoiding this issue)

Tho this fix doesn't solve the `Children name does not match parent name in hashtable, this is a bug.` issue, still not sure how that corruption happens.

Reply:
@RedworkDE Thank you for making this connection!